module/users/helper.js

```diff
@@ -916,6 +973,78 @@ module.exports = class UsersHelper {
   }
 
 
+  /**
+   * Find non-targeted joined program.
+   * @method
+   * @name getUserJoinedProgramDetailsWithPreviousProfiles
+   * @param {Array} targetedProgramIds - programIds
+   * @param {String} searchText - search text
+   * @param {String} userId - userId
+   * @returns {Object} - non-targeted joined program details.
+   */
+  static getUserJoinedProgramDetailsWithPreviousProfiles( targetedProgramIds, searchText = "", userId ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let programUsersIds = [];
+        let nonTargettedProgramDetails = [];
+
+        // find all programs joined by the user
+        // programUsersData will contain list of programs joined by user from all the profiles. This can be considered as the super set of user programs
+        let programUsersData = await programUsersHelper.programUsersDocuments(
+          {
+            userId: userId
+          },
+          ["programId"]
+        );
+        
+        if ( programUsersData.length > 0 ) {
+          programUsersIds = programUsersData.map(function (obj) {
+            return obj.programId;
+          });
+        }
+
+        // if we find the difference between programUsersData and targettedProgramIds we will get program details joined by user other than the current profile
+        let previousProfilesJoinedProgramIds = _.differenceWith(programUsersIds, targetedProgramIds,_.isEqual);
+        
+        if ( previousProfilesJoinedProgramIds.length > 0 ) {
+          let findQuery = {
+            "_id": { "$in" : previousProfilesJoinedProgramIds },
+            "startDate": {"$lte": new Date()},
+            "endDate": {"$gte": new Date()}
+          }
+
+          //call program details to check if the program is active or not
+          let programDetails = await programsHelper.list(
```

Comment:
@VISHNUDAS-tunerlabs Why do we need this check? Is it mentioned in the story that if programs are not active we should not send in this API even if the user has joined?

Reply:
@aks30 by checking active we meant checking startDate and endDate. which we included recently